src/manager/github_runner_manager.py

```diff
@@ -72,14 +72,19 @@ def get_runners(
             Information on the runners.
         """
         runner_list = self.github.get_runner_github_info(self._path)
+        runner_list = [runner for runner in runner_list if runner.name.startswith(self._prefix)]
```

Comment:
I know you have not introduced a change here, but what happens if we have two deployments with the same application name, so perhaps the same unit names, registered to the same organisation? They would share the same prefix. Could this be a problem?

Reply:
Yes, having two different deployments in different juju model, same application name with same unit number, registered on the same repo/org will cause problems. 
We will have to generation a unique string and store it to solve this issue. This was talked about during the early stages of rewrite of this charm, but StoredState was being removed.

Since we have a system to store old configuration as files, I think we can revisit this.

But I think this is out of scope of this PR.